storybook/native/main.js

```diff
@@ -3,10 +3,9 @@ import { AppRegistry } from 'react-native';
 import { getStorybookUI, configure, addDecorator } from '@storybook/react-native';
 import styled, { ThemeProvider } from 'styled-components';
 
+import theme from '../../src/tokens/theme.native';
```

Comment:
you don't need `theme.native` metro will figure out on it's own. Just write 
`import theme from '../../src/tokens/theme';`